### DISCUSSION TITLE
How does Yellow send the mails? What do I have to check with my hoster?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#904] [Ask a question] [answered]
Hello!

I just tried to change the password of the user in the browser (Edit) on a new installation of Yellow.
I get the error message **"Etwas ist schief gelaufen. Can't send email on this server!"**
There is no entry for this in the `yellow-website.log`.
If you don't have the problem, then it's probably my hoster.

How does Yellow send the mails? What do I have to check with my hoster?

**Comment by PetersOtto** [comment]
So, I took a look at the `core.php`. To me it looks like yellow is using the `php mail() function`. Please correct me if I am wrong.
I have asked my hoster if `mail()` is available to me. I am still waiting for an answer.

**Reply by GiovanniSalmeri** [reply]
When I have some time (I hope in October) I will refactor my [Mailer extension](https://github.com/GiovanniSalmeri/yellow-mailer) and adapt it to the new mail API of Yellow. In this way I hope the SMTP (and sendmail) will be available to all other extensions. I take your question as an incitement to hurry up 😊

**Reply by PetersOtto** [reply]
That sounds great! &#128512;

**Reply by GiovanniSalmeri** [reply]
@PetersOtto Now the [Mailer extension](https://github.com/GiovanniSalmeri/yellow-mailer) is updated to the new API and it is automatically used by any other extension which sends mail with `toolbox->mail`. While refactoring Mailer, I have unbundled the example, which now is a separate extension [Multicontact](https://github.com/GiovanniSalmeri/yellow-multicontact). Hope this is useful!

**Reply by PetersOtto** [reply]
Thank you very much for taking care of this! I will test the extension in the next few days.

**Comment by annaesvensson** [comment]
The default way to send emails on Linux is `sendmail`. You can contact your web hoster and ask to enable sendmail. For your information, the core has support for alternative email transports, it's possible to write an extension that sends emails via an alternative transport. SMTP comes to mind. At the moment no developer is working on this, but would be lovely to have something like this in the future. See also previous discussion - warning it's a long one - in datenstrom/community#542.

**Reply by PetersOtto** [reply]
Hello Anna!
Thank you for your answer!
I am glad that Giovanni wants to take care of an SMTP extension. Unfortunately my PHP skills are not sufficient for this.

I checked with `<?php phpinfo();?>` if sendmail is installed. It looks like it is installed.

![sendmail](https://github.com/datenstrom/yellow/assets/52236754/ecd7abfe-ef91-4bfc-9fa9-76d43179f3c7)

I'm still waiting for an answer from the hoster here.

I will have a look to https://github.com/datenstrom/yellow/discussions/542, thank you.

**Reply by annaesvensson** [reply]
Have you received any feedback from your web hoster?

**Reply by PetersOtto** [reply]
Hi Anna.
I have installed my hoster Yellow with their email, so that the hoster can test what the problem is. So far I have not received a info. I will come back to you, when I know more.

**Reply by PetersOtto** [reply]
I have received feedback from my web hoster.
The problem seems, that Yellow try to sent the email without sender. The email goes to the email address of the user that is named in the `yellow-user.ini ` but cannot be sent because no sender email address is named.

Have you an Idea?

**Reply by annaesvensson** [reply, marked as answer]
When you have confirmed that sendmail is enabled, your next option is to change the email address for outgoing emails. The default email address for outgoing emails is `noreply`. The mail server has to add a domain name so that it becomes a complete email address, for example `noreply@example.com`. Now it could be that your mail server is miss-configured or that it adds the wrong domain name to outgoing emails, I don't blame your web hoster, mail servers are notoriously difficult to configure. It's probably easier to change the email address for outgoing emails.

The following settings can be configured in file `system/extensions/yellow-system.ini`:

`ContactSiteEmail` = email address for outgoing emails, [see contact extension](https://github.com/annaesvensson/yellow-contact) 
`EditSiteEmail` = email address for outgoing emails, [see edit extension](https://github.com/annaesvensson/yellow-edit)

**Reply by PetersOtto** [reply]
`EditSiteEmail` works perfectly. 

I didn't think that it could be possible to set an outgoing email address. I should have done better research.
However I still asked my web host if noreply is configured on the server. Maybe the webhoster will find a bug in his configutation.

Thanks for the help!

**Reply by annaesvensson** [reply]
The system settings have been updated, now there's one common setting for outgoing messages:

`From` = email for outgoing messages, [see core extension](https://github.com/annaesvensson/yellow-core)
